Updates docs temporarily to cloudsmith until packages complete

codefromthecrypt opened the pull request:
A month ago, we changed the install page to say official packages are
on the way. #16867 hasn't landed, yet and now problems are surfacing
like #17236, but also spread out over the ecosystem also.

Specifically the "Get Envoy" instructions no longer work because
Bintray no longer works at all and #16867 isn't finished. This fixes the
instructions until packages are available, after which the "Get Envoy"
parts should be deleted as "Get Envoy" is a canceled project due to
trademark issues with the Linux Foundation.

Thanks very much to @carlossg for finding these instructions and
@arkodg for verifying them.

---

These instructions are hand-crafted due to requests from the Envoy
project team. End users can also look at instructions maintained by
Cloudsmith here as they are more likely to remain up-to-date.

Ex. https://cloudsmith.io/~tetrate/repos/getenvoy-deb-stable/setup/#formats-deb

Also note, @phlax insisted these instructions use sudo which is intentionally not installed in docker-based distributions. Those using this with docker will want to install sudo manually or change the instructions accordingly. In any case, the instructions are more manual than necessary to match prior conventions.

arkodg reviewed (approved):
thanks ! LGTM, good interim doc until "Get Envoy" is transitioned out

phlax reviewed (changes requested):
@codefromthecrypt thanks for this - i agree that updating the docs in the interim is probably best

overall i would prefer if you kept the ubuntu section separate

phlax commented on docs/root/start/install.rst:
```diff
@@ -10,51 +10,31 @@ If you are :ref:`installing on Mac OSX <start_install_macosx>`, you can install
-   $ sudo apt install -y getenvoy-envoy
+   $ curl -1sLf \
+       'https://deb.dl.getenvoy.io/public/setup.deb.sh' \
+       | sudo -E bash
```
not sure about this - im very reluctant to run a script from the interwebs as root

looking at the script it seems to require a lot of stuff (eg python) that really shouldnt be necessary to install a deb

codefromthecrypt replied:
sure I'll inline the instructions from the cloud smith site

robejfarr commented:
Hi @codefromthecrypt I'm not sure of the complete history on this and who has ownership of everything, but if I understand it correctly tetrate have moved the publication of rpms from hosting on bintray to cloudsmith.io for now. These features will eventually be upstreamed into Envoy as part of the work @phlax is undertaking.

If the repos referenced in https://rpm.dl.getenvoy.io/public/config.rpm.txt from the shellscript above continue to have the latest rpms published to them then we can simply configure our builds to use these repos and then change again when the work is upstreamed to Envoy. Do you know if these repos will continue to receive the latest envoy updates for the foreseeable future?

codefromthecrypt commented:
@robejfarr Thanks for the feedback. I'll put some stuff here some of it you didn't ask about for other readers

*What's impacted is scoped to deb and RPM "getenvoy" packages*
  * MacOS already has "official" packages (Homebrew), windows never had packages, and "official" docker images were always published.

I think the key facts are:
* These docs are a temporary work around because bintray is down and cloud smith isn't
* We can assume [official packages](https://github.com/envoyproxy/envoy/pull/16899) won't use cloud smith, which means a repo change sooner or later
* Official packages would be named "envoy" not "getenvoy"
* "getenvoy" stops at 1.18.3
* there's no plan to copy and rename official "envoy" packages into cloud smith as "getenvoy"

Since both the repo and also the package names will change, some work will be needed to transition from current packages to official ones.

Folks who are currently impacted will change max twice:
1. temporarily to the "getenvoy" packages on cloud smith repos, func-e.io or a manual process http://archive.tetratelabs.io/envoy
2. to official "envoy" packages complete, manual process on official tarballs #16830, or stay on func-e or Tetrate's archive

Folks who are not currently down can choose what to do and when to do it!

codefromthecrypt commented:
@phlax I switched the instructions to long form and verified them with the corresponding docker images per OS. Note: I removed the "sudo" ops as that's not on the images by default.

robejfarr commented:
Thanks for the update @codefromthecrypt - we're still in development phases with Envoy so we'll switch to the cloudsmith repos for now and review the situation again closer to when we're ready to release. 

I see the 1.19 roadmap has been reduced significantly so the release will be out shortly. Worst case we'll probably package from the binaries to ensure we're getting the latest updates through if official repositories are unavailable prior to our live releases.

phlax commented on docs/root/start/install.rst:
```diff
@@ -13,48 +13,47 @@ getting your Envoy proxy up and running.
-   $ sudo apt install -y getenvoy-envoy
+   $ apt-get update
+   $ apt-get install -y apt-transport-https gnupg2 curl
+   $ curl -1sLf 'https://deb.dl.getenvoy.io/public/gpg.8115BA8E629CC074.key' | apt-key add -
```
the current way of doing this with `deb [signed-by=...]` restricts this key to signing this one archive

codefromthecrypt replied:
feel free to click the suggestion button and apply it or otherwise edit the branch. I'll be AFK for a week and won't have time to look up the implications of what you are saying figure out the translation into a diff then apply it.

codefromthecrypt replied:
@arkodg I just added you as a collab on my fork in case ^^ doesn't happen. Feel free to push anything.

codefromthecrypt replied:
guess this didn't happen which is a shame as I heard people reported known problems. I will follow-up on this personally today

codefromthecrypt commented:
I put a note in the description because while I understand the position of the project team, hand-crafted shell scripts known to have been left broken are less ideal than citing cloudsmith who have support etc. If for some reason, official packages remain unavailable long enough for the shell script snippets to drift again, at least people can see where the supported ones are by looking here.

htuch commented:
@phlax could you take another look? Thanks.

NersesAM commented:
@codefromthecrypt do you maintain the packages on cloudsmith? It doesn't seem to have all the versions of envoy, unlike func-e. Can it be addressed somehow as even if this PR is merged people would expect it to have all versions that were there in bintray at least.

phlax commented on docs/root/start/install.rst:
```diff
@@ -13,48 +13,47 @@ getting your Envoy proxy up and running.
-   $ echo "deb [arch=amd64 signed-by=/usr/share/keyrings/getenvoy-keyring.gpg] https://dl.bintray.com/tetrate/getenvoy-deb $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/getenvoy.list
-   $ sudo apt update
-   $ sudo apt install getenvoy-envoy
+   $ apt-get update
```
can we keep the `sudo` - also minor nit but for enduser use i tend to use `apt` over `apt-get`

phlax commented on docs/root/start/install.rst:
```diff
@@ -13,48 +13,47 @@ getting your Envoy proxy up and running.
-   $ echo "deb [arch=amd64 signed-by=/usr/share/keyrings/getenvoy-keyring.gpg] https://dl.bintray.com/tetrate/getenvoy-deb $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/getenvoy.list
-   $ sudo apt update
-   $ sudo apt install getenvoy-envoy
+   $ apt-get update
```
can we keep the sudo - also minor nit but for enduser use i tend to use apt over apt-get

codefromthecrypt replied:
I removed sudo because it breaks the instructions on the official docker images as it isn't in the path by default. I generally don't like adding instructions that I can't easily verify or require additional steps to operate, but if you feel strongly, I can add it back.

```bash
$ docker run -ti --rm debian
root@2c6dceaf2d6e:/# sudo 
bash: sudo: command not found

$ docker run -ti --rm ubuntu
root@4c41aadc0e0d:/# sudo
bash: sudo: command not found
```

phlax replied:
hmm - so in distros rather than docker images - ubuntu has sudo preinstalled, but debian doesnt

its the ubuntu users im more concerned with tbh - and as a long-time debian user i have reasonable confidence that most debian users can work it out either way

codefromthecrypt replied:
as both sides of this are subjective and the current content is verified. do you mind merging this? I think it is causing more damage and probably we won't hear anyone complain either from lack or presence of sudo. usually nits like this especially subjective ones won't block progress.. let's get this in?

phlax reviewed (changes requested):
@codefromthecrypt apologies for the delay in reviewing - ive been flat out working on the package build

can we add back the sudo - apart from that im happy to land

codefromthecrypt commented:
@NersesAM there's no plan to continue maintenance on the getenvoy packages which are redundant to #16867 and also continues the trademark problems that were behind canceling them.

You can see notes about this for options moving forward. Hopefully official RPMs and debs will land soon. https://github.com/envoyproxy/envoy/pull/17246#issuecomment-876827656

codefromthecrypt commented:
I'm reading silence from @phlax that these instructions must optimize for multi-user distributions of ubuntu, not docker which doesn't install sudo intentionally. I'll edit and redo this again, but please tighten up the feedback loop.

codefromthecrypt commented:
ok please consider merging as it is nearing a month since this PR to fix what should have been fixed 2 months ago

Reverted debian to closest viable version of prior instructions. This required a preliminary step to install sudo, but I left this out of the instructions.
```bash
root@3a70e605649c:/# apt-get update
--snip--
root@3a70e605649c:/# apt-get install sudo
--snip--


### now copy pasting the instructions
root@3a70e605649c:/# sudo apt update
--snip--
root@3a70e605649c:/# sudo apt install debian-keyring debian-archive-keyring apt-transport-https curl lsb-release
--snip--
root@3a70e605649c:/# curl -sL 'https://deb.dl.getenvoy.io/public/gpg.8115BA8E629CC074.key' | sudo gpg --dearmor -o /usr/share/keyrings/getenvoy-keyring.gpg
root@3a70e605649c:/# echo a077cb587a1b622e03aa4bf2f3689de14658a9497a9af2c427bba5f4cc3c4723 /usr/share/keyrings/getenvoy-keyring.gpg | sha256sum --check
/usr/share/keyrings/getenvoy-keyring.gpg: OK
root@3a70e605649c:/# echo "deb [arch=amd64 signed-by=/usr/share/keyrings/getenvoy-keyring.gpg] https://deb.dl.getenvoy.io/public/deb/debian $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/getenvoy.list
--snip--
root@3a70e605649c:/# sudo apt update
--snip--
root@3a70e605649c:/# sudo apt install getenvoy-envoy
--snip--
root@3a70e605649c:/# envoy --version

envoy  version: d362e791eb9e4efa8d87f6d878740e72dc8330ac/1.18.2/clean-getenvoy-76c310e-envoy/RELEASE/BoringSSL

```

Reverted ubuntu to closest viable version of prior instructions. This required a preliminary step to install sudo, but I left this out of the instructions.
```bash
root@63628f4ea374:/# apt-get update
--snip--
root@63628f4ea374:/# apt-get install sudo
--snip--

### now copy pasting the instructions
root@63628f4ea374:/# sudo apt update
--snip--
root@63628f4ea374:/# sudo apt install apt-transport-https gnupg2 curl lsb-release
--snip--
root@63628f4ea374:/# curl -sL 'https://deb.dl.getenvoy.io/public/gpg.8115BA8E629CC074.key' | sudo gpg --dearmor -o /usr/share/keyrings/getenvoy-keyring.gpg
root@63628f4ea374:/# echo a077cb587a1b622e03aa4bf2f3689de14658a9497a9af2c427bba5f4cc3c4723 /usr/share/keyrings/getenvoy-keyring.gpg | sha256sum --check
/usr/share/keyrings/getenvoy-keyring.gpg: OK
root@63628f4ea374:/# echo "deb [arch=amd64 signed-by=/usr/share/keyrings/getenvoy-keyring.gpg] https://deb.dl.getenvoy.io/public/deb/ubuntu $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/getenvoy.list
--snip--
root@63628f4ea374:/# sudo apt update
--snip--
root@63628f4ea374:/# sudo apt install -y getenvoy-envoy 
--snip--
root@63628f4ea374:/# envoy --version

envoy  version: d362e791eb9e4efa8d87f6d878740e72dc8330ac/1.18.2/clean-getenvoy-76c310e-envoy/RELEASE/BoringSSL


```


Reverted centos to closest viable version of prior instructions. This required a preliminary step to install sudo, but I left this out of the instructions.
```bash
[root@f52478579287 /]# yum install sudo     
--snip--


### now copy pasting the instructions
[root@f52478579287 /]# sudo yum install yum-utils
--snip--
[root@f52478579287 /]# sudo rpm --import 'https://rpm.dl.getenvoy.io/public/gpg.CF716AF503183491.key'
[root@f52478579287 /]# curl -sL 'https://rpm.dl.getenvoy.io/public/config.rpm.txt?distro=el&codename=7' > /tmp/tetrate-getenvoy-rpm-stable.repo
[root@f52478579287 /]# sudo yum-config-manager --add-repo '/tmp/tetrate-getenvoy-rpm-stable.repo'
--snip--
[root@f52478579287 /]# sudo yum makecache --disablerepo='*' --enablerepo='tetrate-getenvoy-rpm-stable'
--snip--
[root@f52478579287 /]# sudo yum install getenvoy-envoy 
--snip--
[root@f52478579287 /]# envoy --version

envoy  version: d362e791eb9e4efa8d87f6d878740e72dc8330ac/1.18.2/clean-getenvoy-76c310e-envoy/RELEASE/BoringSSL


```

yanavlasov commented:
Looks like all comments were addressed. @phlax for final pass. I will merge tomorrow if @phlax is busy, since the doc looks correct.

phlax reviewed (approved):
lgtm,  thanks @codefromthecrypt

kanongil commented:
These instructions install an old release (v0.18.2) and not the latest v0.19.0.

codefromthecrypt commented:
@kanongil that's true. there is ongoing work to publish official packages for new releases, but that won't be on a "getenvoy" repository. please see this for a more complete answer https://github.com/envoyproxy/envoy/pull/17246#issuecomment-876827656
